cli/cli_app.go

```diff
@@ -384,7 +384,36 @@ func RunTerragrunt(terragruntOptions *options.TerragruntOptions) error {
 		}
 	}
 
-	return runTerragruntWithConfig(terragruntOptions, terragruntConfig, false)
+	err, output := runTerragruntWithConfig(terragruntOptions, terragruntConfig, false)
+
+	// If running Terraform resulted in an error that matches one of the init errors,
+	// then run the required interim init, and then re-run previous command as-is,
+	// otherwise return error without making any attempt to recover from it or
+	// retry anything.
+	if err != nil && isErrorRequiringInit(output, err, terragruntOptions) {
```

Comment:
I think this routine should only be run if the command is not init. Any errors that `init` should retry on should be captured in the auto-retry feature instead of here.

Reply:
By the "auto-retry" feature, do you mean the `runTerraformWithRetry()`?

Reply:
Yup that's correct!

Reply:
OK. In which case this will work as intended - all the `terraform init` -related error messages are going to be matched against and taken care of in the `runTerraformWithRetry()`.